genic/params.c

```diff
@@ -30,36 +22,43 @@ create_parameters()
     param_declare_string(ps, "OutputDir", REQUIRED, 0, "");
     param_declare_string(ps, "FileBase", REQUIRED, 0, "");
 
-    param_declare_double(ps, "MaxMemoryPerCore", OPTIONAL, 1300., "");
     param_declare_double(ps, "Omega0", REQUIRED, 0, "");
     param_declare_double(ps, "OmegaLambda", REQUIRED, 0, "");
     param_declare_double(ps, "OmegaBaryon", REQUIRED, 0, "");
-    param_declare_int(ps,    "ProduceGas", REQUIRED, 0, "");
-    param_declare_double(ps, "OmegaDM_2ndSpecies", REQUIRED, 0, "");
-    param_declare_int(ps, "UsePeculiarVelocity", OPTIONAL, 0, "Write a IC similiar to a FastPM output");
-    param_declare_double(ps, "HubbleParam", REQUIRED, 0, "");
-    param_declare_double(ps, "ShapeGamma", OPTIONAL, 0.201, "");
-    param_declare_double(ps, "Sigma8", OPTIONAL, -1, "Renoramlize Sigma8 to this number if positive");
-    param_declare_double(ps, "PrimordialIndex", OPTIONAL, 0.971, "Tilting power, ignored for tabulated input.");
+    param_declare_double(ps, "HubbleParam", REQUIRED, 0.7, "Hubble parameter");
+    param_declare_int(ps,    "ProduceGas", REQUIRED, 0, "Should we create baryon particles?");
     param_declare_double(ps, "BoxSize", REQUIRED, 0, "");
-    param_declare_double(ps, "Redshift", REQUIRED, 0, "");
+    param_declare_double(ps, "Redshift", REQUIRED, 99, "Starting redshift");
     param_declare_int(ps, "Nmesh", REQUIRED, 0, "");
-    param_declare_int(ps, "Nsample", REQUIRED, 0, "");
     param_declare_int(ps, "Ngrid", REQUIRED, 0, "");
     param_declare_int(ps, "Seed", REQUIRED, 0, "");
-    param_declare_int(ps, "SphereMode", OPTIONAL, 1, " if 1 only modes with |k| < k_Nyquist are used. otherwise all modes are filled. ");
-    param_declare_int(ps, "WhichSpectrum", OPTIONAL, 2, "Type of spectrum, 2 for file "); 
-    param_declare_double(ps, "UnitVelocity_in_cm_per_s", REQUIRED, 0, "");
-    param_declare_double(ps, "UnitLength_in_cm", REQUIRED, 0, ""); 
-    param_declare_double(ps, "UnitMass_in_g", REQUIRED, 0, "");
-    param_declare_double(ps, "InputSpectrum_UnitLength_in_cm", REQUIRED, 0, "");
-    param_declare_int(ps, "WDM_On", REQUIRED, 0, "");
-    param_declare_int(ps, "WDM_Vtherm_On", REQUIRED, 0, "");
-    param_declare_double(ps, "WDM_PartMass_in_kev", REQUIRED, 0, "");
+    param_declare_int(ps, "WhichSpectrum", REQUIRED, 2, "Type of spectrum, 2 for file ");
+
+    param_declare_double(ps, "MaxMemoryPerCore", OPTIONAL, 1300., "");
+    param_declare_double(ps, "CMBTemperature", OPTIONAL, 2.7255, "CMB temperature in K");
+    param_declare_double(ps, "RadiationOn", OPTIONAL, 1, "Include radiation in the background.");
+    param_declare_int(ps, "UsePeculiarVelocity", OPTIONAL, 0, "Write a IC similiar to a FastPM output");
+    param_declare_double(ps, "Sigma8", OPTIONAL, -1, "Renormalise Sigma8 to this number if positive");
+    param_declare_int(ps, "PowerIsRedshiftZero", OPTIONAL, 1, "If true, rescale the power spectrum by the linear growth factor.");
```

Comment:
What about InputPowerRedshift? Then backscaling is always applied from the inputpower redshift?

Have you thought about taking a transfer function + primordial power input for multiple species IC?